### DISCUSSION TITLE
API 할당된 시간당 요청량을 초과하였습니다. 반환 기준 문의

### DISCUSSION THREAD

**Discussion by kswjjg** [commerce-api-naver/commerce-solutions#82] [묻고 답하기]
솔루션 SELLER 개념으로 주문 API테스트 중에 있는데요.
임의로 천천히 통신하지 않는 한 다건의 주문을 처리할때 '할당된 시간당 요청량을 초과하였습니다' 라는 메세지가 반환되는걸 확인하였습니다.
찾아보니.. 1초당 2건의 통신이 발생하면 해당 이슈가 일어나는 것 같은데, 궁금한게 하나의 셀러 계정으로 여러 서버에서 동시 토큰 발급하여 처리하며, 1초의 딜레이를 주는 방식은 제한이 걸리지 않을까요?
즉 요청량의 제한이 토큰 단위인지 계정 단위인지가 궁금합니다.

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

### 솔루션 애플리케이션의 호출 허용량
솔루션 애플리케이션은 애플리케이션 단위의 API 호출 제한인 "Rate limit"과 SELLER 인증 토큰 단위의 API 호출 제한인 "Quota limit"이 동시에 적용됩니다.
따라서 말씀하신 오류 메시지는 지정된 Quota limit을 초과하는 API 호출이 발생한 경우 반환되었을 것으로 추정됩니다.

커머스API 플랫폼의 호출 허용량은 '토큰 버킷 알고리즘(Token Bucket Rate Limiting Algorithm)'을 활용합니다.
API 호출 제어 부분의 설계는 가급적 시간 딜레이보단 1초당 호출 허용량을 정교하게 제어하는 방식을 권장하고 있습니다.
- 관련 FAQ: https://github.com/commerce-api-naver/commerce-api/discussions/6

### Quota limit의 호출량 산출 기준
Quota limit의 1초당 호출 허용량은 Rate limit 대비 일정 비율로 산정됩니다. Rate limit은 (1)각 API별, (2)대상 솔루션의 사용자 규모별 - 별도로 산정되기에 1초당 Quota limit은 저마다 다를 수 있습니다.
지정된 Quota limit의 초과 호출 여부를 결정짓는 1초당 호출량은 SELLER 인증 토큰별로 집계됩니다. 가령 솔루션 구독자 [A]와 [B]가 각각의 스토어 계정(`account_id`)으로 SELLER 인증 토큰을 받았다면 솔루션 구독자 [A]의 API 호출량은 [B]의 Quota limit 산정에는 영향을 미치지 않습니다. (역방향 영향 역시 성립합니다)
